core/src/main/java/org/infinispan/factories/LockManagerFactory.java

```diff
@@ -11,10 +14,15 @@
  * @author Manik Surtani (<a href="mailto:manik@jboss.org">manik@jboss.org</a>)
  * @since 4.0
  */
-@DefaultFactoryFor(classes = LockManager.class)
+@DefaultFactoryFor(classes = {LockManager.class, PendingLockManager.class} )
 public class LockManagerFactory extends AbstractNamedCacheComponentFactory implements AutoInstantiableFactory {
+   @SuppressWarnings("unchecked")
    @Override
    public <T> T construct(Class<T> componentType) {
+      if (PendingLockManager.class.equals(componentType)) {
+         final boolean clustered = configuration.clustering().cacheMode().isClustered();
+         return clustered ? (T) new DefaultPendingLockManager() : (T) NoOpPendingLockManager.getInstance();
```

Comment:
I know I may be late to the game with this (the unchecked cast is everywhere), but is there any reason we don't use 
```
return componentType.cast(clustered ? new DefaultPendingLockManager() : NoOpPendingLockManager.getInstance());
```